src/OpenLoco/src/Map/TileManager.cpp

```diff
@@ -410,6 +415,21 @@ namespace OpenLoco::World::TileManager
         return height;
     }
 
+    SmallZ getSurfaceCornerHeight(SurfaceElement* surface)
+    {
+        auto baseZ = surface->baseZ();
+        if (surface->slopeCorners() != 0)
```

Comment:
are we sure this is correct shouldn't it also take into account double height. I.e. double height with no slope corners is kSmallZStep*2 (not sure if that is possible though)

Reply:
I'm not sure I understand. Since height is increased in `kSmallZStep` increments, wouldn't that case be covered, anyway?

NB: this code was just extracted from the raise land routine, so the logic should (still) be following vanilla atm.

Reply:
Its subtly different i think. I'm basically wondering if the first if should be `if (surface->slopeCorners() != 0 || surface->isDoubleHeight())`

Reply:
actually there is a function for `(surface->slopeCorners() != 0 || surface->isDoubleHeight())` which is `surface->slope()`